python/cudf/cudf/core/index.py

```diff
@@ -178,6 +178,28 @@ def serialize(self):
     def __contains__(self, item):
         return item in self._values
 
+    @annotate("INDEX_EQUALS", color="green", domain="cudf_python")
+    def equals(self, other):
+        """
+        Determine if two Index objects contain the same elements.
+
+        Returns
+        -------
+        out: bool
+            True if “other” is an Index and it has the same elements
+            as calling index; False otherwise.
+        """
+        if not isinstance(other, Index):
+            return False
```

Comment:
We have this specialization for Index but not for DataFrame or Series. Should we move this to `Frame`?

Reply:
For `DataFrame` and `Series`, we use the hidden kwarg `check_type=True` in `Frame.equals()`, which checks if `type(self) is type(other)`.

For `Index`, we need to allow `IntegerIndex == FloatIndex`, so we set `check_type=False` and check this instead.